### ISSUE TITLE
Docs: Local relative paths not supported by projectService's allowDefaultProject

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#9715]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Repro Code

```TypeScript
(see ESLint config)
```

### ESLint Config

```javascript
// @ts-check

import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';

export default tseslint.config(
  eslint.configs.recommended,
  ...tseslint.configs.recommendedTypeChecked,
  {
    languageOptions: {
      parserOptions: {
        projectService: {
            allowDefaultProject: ['./*.js'],
            defaultProject: 'tsconfig.json'
        },
        tsconfigRootDir: import.meta.dirname,
      },
    },
  },
);
```

### Expected Result

`eslint.config.js` should be picked up by the project service.

### Actual Result

```plaintext
/Users/josh/repos/repros/eslint.config.js
  0:0  error  Parsing error: /Users/josh/repos/repros/eslint.config.js was not found by the project service. Consider either including it in the tsconfig.json or including it in allowDefaultProject
```

### Additional Info

See repro on: https://github.com/JoshuaKGoldberg/repros/tree/repro-allow-default-project-local-relative-path

Followup to #9674 -> #9675: that work made paths like `'*.js'` work, but not `'./*.js'`.

### Versions

| package                                | version  |
| -------------------------------------- | -------- |
| `@typescript-eslint/eslint-plugin`     | `9.8.0`  |
| `@typescript-eslint/parser`            | `9.8.0`  |
| `@typescript-eslint/typescript-estree` | `9.8.0`  |
| `TypeScript`                           | `5.5.4`  |
| `ESLint`                               | `9.8.0`  |
| `node`                                 | `22.4.1` |

💖

**Comment by JoshuaKGoldberg** [comment]
This comes from the internal [`minimatch`](https://www.npmjs.com/package/minimatch) library usage:

```plaintext
const minimatch = require("minimatch");
minimatch("eslint.config.js", "./*.js")
// false
```

We just need to update the docs to use `"*.js"`, not `"./*.js"`.
